### ISSUE TITLE
stat returned ENOENT error for a directory which has symbolic link reference to its starting directory

### ISSUE THREAD

**Issue by anjalirx-intel** [oscarlab/graphene#2578]
## Description of the problem
Accessing the directory which has symbolic link reference to its starting directory, should throw error "Too many levels of symbolic links error (ELOOP) but got ENOENT error.

TC-5:     {loop_dir, ELOOP}

## Steps to reproduce
To repro this issue, run stat03_setup in linux native and stat03_run in Graphene environment. 

## Expected results
 TPASS: stat() failed as expected: ELOOP (40)
 
## Actual results
TFAIL: stat() failed unexpectedly; expected: 40 - ELOOP: ENOENT (2)

[stat03.zip](https://github.com/oscarlab/graphene/files/6867987/stat03.zip)

**Comment by anjalirx-intel** [comment]
@pwmarcz Hi Pawel, 
Currently, symlink is not supported in Graphene(ENOSYS), so to test ELOOP error for stat function, we have split the code and created the symlink and directory in linux environment and when tested with Graphene the stat returns ENOENT error.

Is there a way we can create symlink inside Graphene itself?

**Comment by pwmarcz** [comment]
Hi. Currently there is no way to create your own symlink in Graphene; there are only symlinks in "pseudo" filesystems (such as `/proc/self -> /proc/PID`).

I'm planning to implement the `symlink` call: we talked about it with other maintainers, and we want it to create in-memory symlinks (same as we do with named pipes and sockets). However, right now this is not a priority, because AFAIK there is no application blocked by lack of symlink support.
